modin/numpy/math.py

```diff
@@ -14,16 +14,16 @@
 import numpy
 
 from .arr import array
+from .utils import try_convert_from_interoperable_type
 from modin.error_message import ErrorMessage
 
 
 def absolute(
     x, out=None, where=True, casting="same_kind", order="K", dtype=None, subok=True
 ):
+    x = try_convert_from_interoperable_type(x)
     if not isinstance(x, array):
-        ErrorMessage.single_warning(
-            f"Modin NumPy only supports objects of modin.numpy.array types for absolute, not {type(x)}. Defaulting to NumPy."
-        )
+        ErrorMessage.bad_type_for_numpy_op("absolute", type(x))
```

Comment:
I wonder if this should be hidden in a decorator, like this (untested):
```python
from functools import wraps

def _default_to_numpy(func):
    @wraps(func):
    def wrapped(arr, *args, **kw):
        arr = try_convert_from_interoperable_type(arr)
        if not isinstance(arr, array):
            ErrorMessage.bad_type_for_numpy_op(func.__name__, type(arr))
            return getattr(numpy, func.__name__)(arr, *args, **kw)
        return func(arr, *args, **kw)
    return wrapped


@_default_to_numpy
def absolute(
    x, out=None, where=True, casting="same_kind", order="K", dtype=None, subok=True
):
    return x.absolute(
        out=out, where=where, casting=casting, order=order, dtype=dtype, subok=subok
    )

@_default_to_numpy
def add(
    x1, x2, out=None, where=True, casting="same_kind", order="K", dtype=None, subok=True
):
    return x1.__add__(
        x2,
        out=out,
        where=where,
        casting=casting,
        order=order,
        dtype=dtype,
        subok=subok,
    )

# etc.
```
?

Reply:
Perhaps - I think its little enough code that its fine for now IMO!

Reply:
I guess this can be handled during the refactor you and @noloerino were planning ahead...